onnxruntime/core/providers/dml/DmlExecutionProvider/src/GraphicsUnknownHelper.h

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+namespace Dml
+{
+// D3D12.x interfaces inherit from IGraphicsUnknown, which does not inherit from IUnknown. This
+// wrapper exists to pass IGraphicsUnknown-inheriting objects to functions with IUnknown parameters.
+#ifdef _GAMING_XBOX
+class GraphicsUnknownWrapper : public Microsoft::WRL::RuntimeClass<Microsoft::WRL::RuntimeClassFlags<Microsoft::WRL::ClassicCom>, IUnknown>
+{
+public:
+    explicit GraphicsUnknownWrapper(const Microsoft::WRL::ComPtr<IGraphicsUnknown>& graphicsUnknown) : m_graphicsUnknown(graphicsUnknown.Get()) {}
```

Comment:
>const Microsoft::WRL::ComPtr<IGraphicsUnknown>& graphicsUnknown [](http://example.com/codeflow?start=36&length=63)

Taking this by value and using std::move can save an extraneous ref in some cases.